### PR TITLE
Bug/219/madx macros and more

### PR DESCRIPTION
**Pull request by mihofer**
Moving lib content to model/madx_macros
Include non .py files when installing with `pip install git+https://github.com/pylhc/omc3.git`

Change test handling

Remove python 3.6 from tests and requirements, after merge only 3.7 and above is maintained, corrdinated merge with new tfs

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -6,33 +6,6 @@
 from setuptools.command.test import test as TestCommand
```
Comment (fsoubelet) [inline setup.py:6]: Shouldn't be needed anymore

Comment (mihofer) [reply setup.py]: fixed

**Comment by fsoubelet** [comment]
Do we need a `pytest.ini` just to describe two markers?

**Comment by mihofer** [comment]
> 
> 
> Do we need a `pytest.ini` just to describe two markers?

Kinda yes, kinda no .. if the markers are not specified, it will complain every time that it could be wrong and running it with the --strict-markers would fail then ( https://docs.pytest.org/en/stable/mark.html ) .. one could also define it somewhere programmatically but pytest.ini looked more convenient to me

**Comment by fsoubelet** [comment]
Fair enough. Wouldn't we want to group most tests in verbosely named classes and `mark` the classes to avoid getting decorators everywhere? Grouping test functions in classes around common themes would also let us easily test a single thematic with the `-k` option from `pytest`.

Additionally, does calling `pytest -m basic` also run functions that aren't decorated? If not, maybe we should change the CI command to `pytest -m "not basic"` in `.travis.yml` to ensure that branch builds run everything except the extended tests.

**Review by fsoubelet — CHANGES_REQUESTED** [review]
Can you remove `pytest-runner` and `setup_requires` as discussed?

Seems to me the `shlex` and `sys` imports aren't needed anymore in `setup.py`.

**Comment by mihofer** [comment]
> 
> 
> Can you remove `pytest-runner` and `setup_requires` as discussed?
> 
> Seems to me the `shlex` and `sys` imports aren't needed anymore in `setup.py`.

`pytest-runner` remains should be gone now and unused imports removed

**Review by fsoubelet — DISMISSED** [review]

**Comment by fsoubelet** [comment]
Seems all good to me. Should rerun builds when `tfs-pandas` is ready for `2.0` to check that all works fine, and we'll be good.

**Comment by fsoubelet** [comment]
Actually I don't think we need `hypothesis` and `attrs` as test dependencies either @mihofer

**Comment by lmalina** [comment]
> Actually I don't think we need `hypothesis` and `attrs` as test dependencies either @mihofer

`attrs` is not our direct dependency, however tests in travis wouldn't run without it

**Comment by fsoubelet** [comment]
@mihofer do we need all the `@pytest.mark.basic` markers actually? Considering that for "basic" tests we run `-m 'not extended'`

**Comment by mihofer** [comment]
> 
> 
> @mihofer do we need all the `@pytest.mark.basic` markers actually? Considering that for "basic" tests we run `-m 'not extended'`

as it is now no, though I think for completeness it's not bad to mark every test .. seeing issue #228 we could then also add a `fast` marker to certain tests

File: setup.cfg [hunk 2]
```diff
@@ -3,7 +3,9 @@ codestyle_max_line_length = 100
 codestyle_exclude = doc/conf.py
 addopts = --cov-report xml --cov=omc3
 testpaths = tests/unit tests/accuracy
-python_classes = BasicTests ExtendedTests
+markers=
+        basic: basic tests run for every commit
+        extended: test run on PRs
```
Comment (lmalina) [inline setup.cfg:8]: just one of them should be enough

File: setup.py [hunk 3]
```diff
@@ -50,11 +20,11 @@ def run_tests(self):
 DEPENDENCIES = [
     "matplotlib>=3.2.0",
     "Pillow>=6.2.2",  # not our dependency but older versions crash with mpl
-    "numpy>=1.18.0",
-    "pandas==0.25.*",
-    "scipy>=1.4.0",
+    "numpy>=1.19.0",
+    "pandas>=1.0",
+    "scipy>=1.5.0",
     "scikit-learn>=0.22.0",
```
Comment (lmalina) [inline setup.py:26]: scikit-learn>=0.23.0

**Review by lmalina — CHANGES_REQUESTED** [review]

File: setup.py (hunk 3 above)
Comment (mihofer) [reply setup.py:26]: more information plz? i hope you didnt type it on your smartphone so a bit more writing can be expected ...

File: setup.cfg (hunk 2 above)
Comment (mihofer) [reply setup.cfg:8]: should be enough but there is also no solid reason why to scrape it, is there? and anybody in the future might appreciate to find a reason why we distinguish

File: omc3/madx_wrapper.py [hunk 4]
```diff
@@ -22,7 +22,7 @@
 
 LOG = logging_tools.get_logger(__name__)
 
-LIB = abspath(join(dirname(__file__), "lib"))
+LIB = abspath(join(dirname(__file__), "model", "madx_macros"))
```
[the remark this answers is not in the document]
Comment (mihofer) [reply omc3/madx_wrapper.py]: moved to tests

File: setup.py (hunk 3 above)
Comment (lmalina) [reply setup.py]: first version compatible with python 3.8

**Review by lmalina — APPROVED** [review]

File: setup.cfg [hunk 5]
```diff
@@ -3,7 +3,9 @@ codestyle_max_line_length = 100
 codestyle_exclude = doc/conf.py
```
Comment (fsoubelet) [inline setup.cfg:3]: Delete lines 2 and 3 since they're meant for `pycodestyle`

File: setup.py [hunk 6]
```diff
@@ -50,11 +20,11 @@ def run_tests(self):
 DEPENDENCIES = [
     "matplotlib>=3.2.0",
     "Pillow>=6.2.2",  # not our dependency but older versions crash with mpl
-    "numpy>=1.18.0",
-    "pandas==0.25.*",
-    "scipy>=1.4.0",
-    "scikit-learn>=0.22.0",
-    "tfs-pandas>=1.0.3",
+    "numpy>=1.19.0",
+    "pandas>=1.0",
```
Comment (fsoubelet) [inline setup.py:24]: Do really we need to force pandas `>=1.0`?

Leaving `>=0.25` instead will let `pip` install `1.x` which we're compatible with but give some more freedom in environments.

**Review by fsoubelet — DISMISSED** [review]

File: setup.py (hunk 6 above)
Comment (mihofer) [reply setup.py:24]: But new tfs requirements are also >=1.0 so  i think there is nothing to be gained there as well, is there?

Comment (fsoubelet) [reply setup.py:24]: I guess you're right. @lmalina do you approve?

Comment (lmalina) [reply setup.py:24]: we need pandas >=1.0 because of .convert_dtypes()

**Review by lmalina — APPROVED** [review]

**Review by fsoubelet — APPROVED** [review]

**Comment by fsoubelet** [comment]
Good to me 👍

**Pull request merged**